Introducing a nudge-bot to help with triaging issues

sureshjoshi opened the pull request.

cognifloyd commented:
When I'm testing a workflow_dispatch workflow, I:
- use a fork,
- enable actions for the fork,
- update any steps that check the org name to check for the fork instead,
- add the target workflow (or at least a basic version of it) on the default branch or change the fork's default branch so that the default branch includes a workflow file with the same name as the target workflow and that is configured to run on workflow_dispatch. (You can also just set the default branch to the branch you're developing in)
- trigger the workflow manually, to test it, making sure to select the branch where you are editing the workflow file -- that could be the fork's default branch or another branch.

cognifloyd commented on .github/workflows/nudgebot.yaml:
```diff
@@ -0,0 +1,37 @@
+jobs:
+  close-issues:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == "pantsbuild"
```
This makes testing in forks annoying, because you have to change it or comment it out to get the workflow running?

sureshjoshi commented:
@cognifloyd  Thanks - this is a weird one, since we rely on really old tickets, so I think I'll make a branch in the pants repo and use the "Branch" setting of github actions, feels like a PR from a fork isn't the move when I want to triage old issues

sureshjoshi commented:
Example output: 

```
Statistics:
Processed issues: 646
New stale issues: 249
Added issues labels: 249
Added issues comments: 249
Fetched items: 700
Fetched items events: 249
Fetched items comments: 249
Operations performed: 1003
##[debug]Github API rate status: limit=5000, used=635, remaining=4365
Github API rate used: 505
Github API rate remaining: 4365; reset at: Tue Oct 21 2025 15:45:36 GMT+0000 (Coordinated Universal Time)
Warning: The state is not persisted in the debug mode
```

cognifloyd commented on .github/workflows/nudgebot.yaml:
```diff
@@ -0,0 +1,37 @@
+jobs:
+  label-inactive-issues:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'pantsbuild'
```
This makes testing in forks difficult, but, like you said, this is dealing with issues in the main repo, so testing in a fork might not provide much value.
